src/Share/Survey/SurveyHelper.cs

```diff
@@ -90,7 +90,7 @@ public bool ShouldPropmtSurvey(string moduleName, Version moduleVersion)
 
             if (Modules.Count == 0)
             {
-                ReadFromStream();
+                if (!ReadFromStream()) { return false; }
```

Comment:
Previously, if ReadFromStream returned false (this may happen in multi-process), the Modules dictionary would be empty, and ShouldModulePrompt would throw a key-not-found exception when trying to access the dictionary.